system/ee/ExpressionEngine/Addons/structure/tab.structure.php

```diff
@@ -718,9 +718,14 @@ public function get_parent_fields($entry_id, $data)
 
         // PARENT BUG
         // Update: Changed this to `$entry_id` which may cause some oddity with the Parent dropdown.
-        $data = $this->sql->get_data($entry_id);
+        // $data = $this->sql->get_data($entry_id);
 
         foreach ($data as $eid => $entry) {
+            // If we have an entry then this entry and its children
+            // cannot be an option to use as its own parent
+            if ($entry_id && ($eid == $entry_id || $entry['parent_id'] == $entry_id)) {
+                continue;
+            }
```

Comment:
This does not work for entries that are grand-children and grand-grand-children, and it causing weird formatting in the dropdown.
While some cleanup would be good idea, this particular approach is not working 100% and as such I don't think it should be in.